### DISCUSSION TITLE
What does `-> torch.Tensor` mean?

### DISCUSSION THREAD

**Discussion by mostfa00** [mrdbourke/pytorch-deep-learning#198] [Q&A] [answered]
In the code: 

```python
class LinearRegressionModel(nn.Module):
    def __init__(self):
        super().__init__()
        self.weights = nn.Paramter(torch.randn(1,
                                               requires_grad=True,
                                               dtype =torch.float))
        
        self.bias = nn.Paramter(torch.randn(1,
                                            requires_grad=True,
                                            dtype =torch.float))
        
        def forward(self, x: torch.Tensor) -> torch.Tensor:
            return self.weights * x + self.bias
```

What is the meaning of  `x: torch.Tensor) -> torch.Tensor` in last line?

**Comment by fivefishstudios** [comment, marked as answer]
They're called function typehints. i.e. the output/return value of that function forward() is a type of 'torch.Tensor' 

I think it's only used for documentation and code hints and not actually enforced by Python.

**Comment by mrdbourke** [comment]
Thank you @fivefishstudios for the answer! 

And @mostfa00, they aren't required but they do help with code/communication.

Type hints are a way of specifying the expected data types of function arguments and return values in Python.

Here's an example:

```python
from typing import List

def combine_strings(strings: List[str]) -> str:
    return ''.join(strings)
```

In this example, the `combine_strings()` function takes a list of strings as an argument and returns a single string that is the concatenation of all the strings in the list. 

The `List[str]` and `str` type hints specify that the function takes a list of strings as its argument and returns a string (`-> str` is like saying "returns a string").
